torchtune/_cli/download.py

```diff
@@ -159,3 +202,91 @@ def _download_cmd(self, args: argparse.Namespace) -> None:
+            output_dir = Path("/tmp") / Path(model_name)
+
+        if bool(args.kaggle_username) ^ bool(args.kaggle_api_key):
+            self._parser.error(
+                "Both your Kaggle username and api key are needed to authenticate to Kaggle, but only one was provided."
+            )
+        elif bool(args.kaggle_username) and bool(args.kaggle_api_key):
+            set_kaggle_credentials(args.kaggle_username, args.kaggle_api_key)
+
+        try:
+            kaggle_output_path = model_download(model_handle)
+        except KaggleApiHTTPError as e:
+            if e.response.status_code in {
+                HTTPStatus.UNAUTHORIZED,
+                HTTPStatus.FORBIDDEN,
+            }:
+                self._parser.error(
+                    "It looks like you are trying to access a gated model. Please ensure you "
+                    "have access to the model and have provided the proper Kaggle credentials "
+                    "using the options `--kaggle-username` and `--kaggle-api-key`. You can also "
+                    "set these to environment variables as detailed in "
+                    "https://github.com/Kaggle/kagglehub/blob/main/README.md#authenticate."
+                )
+            elif e.response.status_code == HTTPStatus.NOT_FOUND:
+                self._parser.error(
+                    f"'{model_handle}' not found on the Kaggle Model Hub."
+                )
+            tb = traceback.format_exc()
+            msg = f"Failed to download {model_handle} with error: '{e}' and traceback: {tb}"
+            self._parser.error(msg)
+        except Exception as e:
+            tb = traceback.format_exc()
+            msg = f"Failed to download {model_handle} with error: '{e}' and traceback: {tb}"
+            self._parser.error(msg)
+
+        try:
+            # manually copy contents as workaround until kagglehub supports explicit output_dir specification
+            true_output_dir = copytree(
+                kaggle_output_path, output_dir, dirs_exist_ok=True
+            )
+        except Exception as e:
+            tb = traceback.format_exc()
+            msg = f"Failed to copy {model_handle} from {kaggle_output_path} to {output_dir} with error: '{e}' and traceback: {tb}"
+            warn(msg)
+
+            # stick with the path originally written to by kagglehub
+            true_output_dir = kaggle_output_path
+
+        # since kagglehub may output a single filepath, normalize this to a list of files
+        if os.path.isdir(true_output_dir):
+            output_list = list(Path(true_output_dir).iterdir())
+        else:
+            output_list = [true_output_dir]
+
+        print(
+            "Successfully downloaded model repo and wrote to the following locations:",
+            *output_list,
+            sep="\n",
+        )
+
+    def _validate_kaggle_model_handle(self, handle: str) -> None:
+        try:
+            handle_parts = handle.split("/")
```

Comment:
Do you want to use [this](https://github.com/Kaggle/kagglehub/blob/main/src/kagglehub/handle.py#L115) instead?